sphericart-torch/include/sphericart/cuda.hpp

```diff
@@ -1,26 +1,26 @@
 #ifndef SPHERICART_TORCH_CUDA_HPP
 #define SPHERICART_TORCH_CUDA_HPP
 
-#include <ATen/Tensor.h>
-
-#include <vector>
-
 namespace sphericart_torch {
 
-bool adjust_cuda_shared_memory(at::ScalarType scalar_type, int64_t l_max,
+template <typename scalar_t>
+void spherical_harmonics_cuda_base(
```

Comment:
Is this the main function for users who want to use sphericart CUDA without going through torch?

If so, should we move it out of `sphericart-torch` and into `sphericart`?

Reply:
This function needs documentation

Reply:
yeah this is the main function. We should move cuda_base.cu to sphericart and have only the torch wrapper (torch_cuda_wrapper.cpp) present in sphericart-torch.

The documentation for spherical_harmoncis_cuda_base is already in the .cu file:

/*
    Wrapper to launch the CUDA kernel. Returns a vector containing the spherical
   harmonics and their gradients if required, otherwise returns the spherical
   harmonics and an empty tensor.

    GRID_DIM_X is the number of threads to launch in the x dimension. Used to
   parallelize over the sample dimension. GRID_DIM_Y is the number of threads to
   launch in the y dimension. Used only to improve memory throughput on reads
   and writes.

    Total number of threads used is GRID_DIM_X * GRID_DIM_Y.
*/

Reply:
ok, we should move the documentation to the header then, since that's where users typically expect to find it. The snippet above should also be expanded with a description of each parameter.

Reply:
Fixed this in the latest commit.